src/sql/workbench/services/notebook/common/sqlSessionManager.ts

```diff
@@ -305,10 +310,13 @@ export class SQLFuture extends Disposable implements FutureInternal {
 	}
 
 	public handleDone(err?: Error): void {
-		let msg: nb.IShellMessage = {
+		let msg: nb.IExecuteReplyMsg = {
```

Comment:
Totally makes sense to change this. Just making sure that this doesn't break any scenarios (running the same cell multiple times, running a few cells, etc), since it's a different message type.

Reply:
So - this is how the other implementations do it (they always case to IExecuteReplyMsg). It's a subclass of IShellMessage and then in our handler I check that it's actually a execute message before retrieving the data we want (execution_count).